brokerapi/brokers/gcp_service_broker.go

```diff
@@ -342,23 +348,21 @@ func (gcpBroker *GCPServiceBroker) Unbind(ctx context.Context, instanceID, bindi
 		"details":     details,
 	})
 
+	service := gcpBroker.ServiceBrokerMap[details.ServiceID]
+
 	// validate existence of binding
-	var count int
 	existingBinding := models.ServiceBindingCredentials{}
-
-	if err := db_service.DbConnection.Where("service_instance_id = ? and binding_id = ?", instanceID, bindingID).Find(&existingBinding).Count(&count).Error; err != nil {
+	if err := db_service.DbConnection.Where("service_instance_id = ? and binding_id = ?", instanceID, bindingID).Find(&existingBinding).Error; err != nil {
```

Comment:
Is this an edge case not covered by the DAO?

Reply:
It's not right now because it's a fake composite primary key. It should be labeled as the primary key and unique in the database, but for some reason the authors used it as a separate PK field and didn't even add a constraint to enforce uniqueness on this.

I left the implementation of the templates open to allow for generating queries with multiple keys, but didn't need it for the base cases I was covering at the time: https://github.com/GoogleCloudPlatform/gcp-service-broker/blob/master/db_service/dao_generator.go#L145

Reply:
Would it be worthwhile to create an issue to enable support for composite PKs like this? Perhaps leaving a TODO comment here referencing the issue.

Reply:
I was planning on tracking this as part of #216, this PR was me doing ~1/4 of the work in the easiest function to refactor to see if I missed anything big before I got into doing the other 3 (provision, bind, unbind).

Reply:
Sounds good.